src/validator/GroupByValidator.cpp

```diff
@@ -34,38 +34,55 @@ Status GroupByValidator::validateYield(const YieldClause* yieldClause) {
 
     projCols_ = qctx_->objPool()->add(new YieldColumns);
     for (auto* col : columns) {
-        auto rewrited = false;
         auto colOldName = deduceColName(col);
+        auto* colExpr = col->expr();
         if (col->expr()->kind() != Expression::Kind::kAggregate) {
-            NG_RETURN_IF_ERROR(rewriteInnerAggExpr(col, rewrited));
+            auto collectAggCol = colExpr->clone();
+            auto aggs =
+                ExpressionUtils::collectAll(collectAggCol.get(), {Expression::Kind::kAggregate});
+            for (auto* agg : aggs) {
+                DCHECK_EQ(agg->kind(), Expression::Kind::kAggregate);
+                if (!ExpressionUtils::checkAggExpr(static_cast<const AggregateExpression*>(agg))
+                         .ok()) {
+                    return Status::SemanticError("Aggregate function nesting is not allowed: `%s'",
+                                                 colExpr->toString().c_str());
+                }
+
+                groupItems_.emplace_back(qctx_->objPool()->add(agg->clone().release()));
+                needGenProject_ = true;
+                outputColumnNames_.emplace_back(agg->toString());
+            }
+            if (aggs.size()) {
+                auto* colRewrited = ExpressionUtils::rewriteAgg2VarProp(colExpr);
+                projCols_->addColumn(new YieldColumn(colRewrited, new std::string(colOldName)));
+                projOutputColumnNames_.emplace_back(colOldName);
+                continue;
+            }
         }
-        auto colExpr = col->expr();
+
         // collect exprs for check later
         if (colExpr->kind() == Expression::Kind::kAggregate) {
             auto* aggExpr = static_cast<AggregateExpression*>(colExpr);
             NG_RETURN_IF_ERROR(ExpressionUtils::checkAggExpr(aggExpr));
-        } else {
+        } else if (!ExpressionUtils::isEvaluableExpr(colExpr)) {
```

Comment:
Why we have this restriction？

Reply:
Avoid constant expressions as `groupKey` when doing implicit groupby.